### DISCUSSION TITLE
Button to enable/disable auto-switching for all profiles

### DISCUSSION THREAD

**Discussion by mathewharwich** [elgatosf/elgato#136] [Ideas]
I use stream deck for multiple software. I have set the profiles to open automatically with that software. This is great... most of the time. However, there are so many times where this is a bad thing, like when I'm using multiple software at once (like obs for instance). I want to be able to program a button that will keep a profile pinned on the streamdeck even if I switch to another application. The button programming options could have a dropdown to select which profile this is, or maybe just be able to keep the 'current' profile pinned or active. Or---- just a way to "disable auto-profile switching" And it would be even better if this button could be a toggle by default.

The only way currently to switch off auto switching the profiles is to disable them all one by one, and that is also bad because then I have to do everything fully manually. All we need is a switch to turn off profile switching or pin the current profile, or whatever you want to call it. This function is essential I think for stream deck to be excellent. Especially for OBS (but should be available for all profiles). Consider that when users are recording or streaming with OBS, they are using other software, so the stream decks profile will switch if they have an auto-switching profile on for that software, and that might be bad, because we might want to have those stream decks buttons available to control obs while still in the other software. However, we all want to be able to have auto-switching on and available for all those other times where it's favorable. I think it would be a win-win situation to add this simple little functionality to the stream deck. Simple stream deck plugin button to add and I could see this in one of the general default plugins of OBS (such as in Navigation, where other buttons like "switch profile" and "go to page" are. Could be called:
-pin/unpin current profile (toggle)
or you could call it
-enable/disable auto-switching (of all profiles)

I'm not a developer, and I'm not sure this functionality is programmable by 3rd parties. Hoping this idea gets into the right hands, Elgato or otherwise.

**Comment by ChekTek** [comment]
Hi @mathewharwich, 

Unfortunately, this is not possible via our SDK available for plugins, but I will take this feature request to the Stream Deck core team! 

In the meantime, have you tried pinning specific actions to a page? I understand this doesn't exactly accomplish what you are looking to do with profiles, but you could potentially setup a page per application and rotate through them manually. Or copy your actions from profile to profile. 

![CleanShot 2024-10-10 at 14 41 42](https://github.com/user-attachments/assets/fe952902-c243-4440-854e-5095e11af05b)

Best,
Zack
DevRel @Elgatosf
